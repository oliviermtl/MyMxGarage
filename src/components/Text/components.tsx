import { StyleSheet } from 'react-native';
import Parsed from 'react-native-parsed-text';
import styled, { css } from 'styled-components/native';
import { fontFamilies, scaleFontSize } from '../../constants';

const commonTextProps = css`
  width: ${({ width }) => (width ?? width === null ? 'auto' : '100%')};
  text-align: ${({ align }) => align};
  color: ${({ color }) => color};
  margin-top: ${(props) => props.marginTop || 0}px;
  ${({ paddingHorizontal, paddingLeft }) =>
    (paddingHorizontal && paddingLeft) || paddingHorizontal
      ? css`
          padding-left: ${paddingHorizontal}px;
          padding-right: ${paddingHorizontal}px;
        `
      : paddingLeft
      ? css`
          padding-left: ${paddingLeft}px;
        `
      : ''}
`;

export const H1 = styled.Text`
  font-size: ${scaleFontSize(24)}px;
  line-height: ${scaleFontSize(30)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'bold']};
  ${commonTextProps}
`;

export const H2 = styled.Text`
  font-size: ${scaleFontSize(20)}px;
  line-height: ${scaleFontSize(25)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'demiBold']};
  ${commonTextProps}
`;

export const H3 = styled.Text`
  font-size: ${scaleFontSize(17)}px;
  line-height: ${scaleFontSize(21)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'bold']};
  ${commonTextProps}
`;

export const H4 = styled.Text`
  font-size: ${scaleFontSize(12)}px;
  line-height: ${scaleFontSize(18)}px;
  letter-spacing: ${scaleFontSize(4)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'demiBold']};
  ${commonTextProps}
`;

export const P1 = styled.Text`
  font-size: ${scaleFontSize(16)}px;
  line-height: ${scaleFontSize(22)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'regular']};
  ${commonTextProps}
`;

export const P2 = styled.Text`
  font-size: ${scaleFontSize(14)}px;
  line-height: ${scaleFontSize(20)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'regular']};
  ${commonTextProps}
`;

export const P3 = styled.Text`
  font-size: ${scaleFontSize(14)}px;
  line-height: ${scaleFontSize(18)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'demiBold']};
  ${commonTextProps}
`;

export const C1 = styled.Text`
  font-size: ${scaleFontSize(13)}px;
  line-height: ${scaleFontSize(18)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'regular']};
  ${commonTextProps}
`;

export const C2 = styled.Text`
  font-size: ${scaleFontSize(12)}px;
  line-height: ${scaleFontSize(15)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'demiBold']};
  ${commonTextProps}
`;

export const Form = styled.Text`
  font-size: ${scaleFontSize(16)}px;
  line-height: ${scaleFontSize(22)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'regular']};
  ${commonTextProps}
`;

export const Button1 = styled.Text`
  font-size: ${scaleFontSize(16)}px;
  line-height: ${scaleFontSize(20)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'bold']};
  ${commonTextProps}
`;

export const Button2 = styled.Text`
  font-size: ${scaleFontSize(14)}px;
  line-height: ${scaleFontSize(18)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'bold']};
  ${commonTextProps}
`;

export const Button3 = styled.Text`
  font-size: ${scaleFontSize(12)}px;
  line-height: ${scaleFontSize(15)}px;
  font-family: ${({ fontFamily }) => fontFamilies[fontFamily ?? 'bold']};
  ${commonTextProps}
`;

export const ParsedText = styled(Parsed)`
  font-size: ${scaleFontSize(14)}px;
  text-align: center;
  letter-spacing: 0.2px;
  color: ${(props) => props.color};
  font-family: '${fontFamilies.regular}';
`;

export const textStyles = StyleSheet.create({
  H1: {
    fontSize: scaleFontSize(24),
    lineHeight: scaleFontSize(30),
    fontFamily: fontFamilies.bold,
  },
  H2: {
    fontSize: scaleFontSize(20),
    lineHeight: scaleFontSize(25),
    fontFamily: fontFamilies.demiBold,
  },
  H3: {
    fontSize: scaleFontSize(17),
    lineHeight: scaleFontSize(21),
    fontFamily: fontFamilies.bold,
  },
  H4: {
    fontSize: scaleFontSize(12),
    lineHeight: scaleFontSize(18),
    letterSpacing: scaleFontSize(4),
    fontFamily: fontFamilies.demiBold,
  },
  P1: {
    fontSize: scaleFontSize(16),
    lineHeight: scaleFontSize(22),
    fontFamily: fontFamilies.regular,
  },
  P2: {
    fontSize: scaleFontSize(14),
    lineHeight: scaleFontSize(20),
    fontFamily: fontFamilies.regular,
  },
  P3: {
    fontSize: scaleFontSize(14),
    lineHeight: scaleFontSize(18),
    fontFamily: fontFamilies.demiBold,
  },
  C1: {
    fontSize: scaleFontSize(13),
    lineHeight: scaleFontSize(16),
    fontFamily: fontFamilies.demiBold,
  },
  C2: {
    fontSize: scaleFontSize(12),
    lineHeight: scaleFontSize(15),
    fontFamily: fontFamilies.demiBold,
  },
  Form: {
    fontSize: scaleFontSize(16),
    lineHeight: scaleFontSize(22),
    fontFamily: fontFamilies.regular,
  },
  Button1: {
    fontSize: scaleFontSize(16),
    lineHeight: scaleFontSize(20),
    fontFamily: fontFamilies.bold,
  },
  Button2: {
    fontSize: scaleFontSize(14),
    lineHeight: scaleFontSize(18),
    fontFamily: fontFamilies.bold,
  },
});
