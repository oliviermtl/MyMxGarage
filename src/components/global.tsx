import LottieView from 'lottie-react-native';
import {
  ActivityIndicator,
  Dimensions,
  ImageSourcePropType,
  StyleSheet,
  ViewStyle,
  TouchableOpacityProps,
  StyleProp,
  ImageProps,
} from 'react-native';
import { hasNotch } from 'react-native-device-info';
import FastImage from 'react-native-fast-image';
import Spinner from 'react-native-loading-spinner-overlay';
import styled, { css } from 'styled-components/native';
import { scaleFontSize, fontFamilies, colors } from '../constants';

const { width: viewportWidth } = Dimensions.get('window');

export type numberStringNull = number | string | null;

export const globalWidth = viewportWidth - 32;
export const rowWidth = globalWidth;
export const singleFieldWidth = rowWidth / 2 - 8;

export const TouchableOpacity = styled.TouchableOpacity<{ width?: numberStringNull }>`
  width: ${({ width = 'auto' }) => width};
`;
export const TouchableOpacityDefault = styled.TouchableOpacity``;
export const TouchableWithoutFeedback = styled.TouchableWithoutFeedback``;
export const Pressable = styled.Pressable.attrs({ unstable_pressDelay: 100 })``;

interface ViewProps {
  height?: numberStringNull;
  width?: numberStringNull;
  backgroundColor?: string;
  alignItems?: string;
  justifyContent?: string;
  padding?: number;
  paddingHorizontal?: number;
  paddingVertical?: number;
  marginTop?: number;
  marginBottom?: number;
  flex?: number | string;
}

export const View = styled.View<ViewProps>`
  height: ${({ height = 'auto' }) => height};
  width: ${({ width = 'auto' }) => width};
  background-color: ${({ backgroundColor = 'transparent' }) => backgroundColor};
  align-items: ${({ alignItems = 'flex-start' }) => alignItems};
  justify-content: ${({ justifyContent = 'flex-start' }) => justifyContent};
  padding-horizontal: ${({ paddingHorizontal = 0 }) => paddingHorizontal}px;
  padding-vertical: ${({ paddingVertical = 0 }) => paddingVertical}px;
  margin-top: ${({ marginTop = 0 }) => marginTop}px;
  margin-bottom: ${({ marginBottom = 0 }) => marginBottom}px;
  ${({ flex }) =>
    flex &&
    css`
      flex: ${flex};
    `}
`;
export const DefaultView = styled.View<ViewProps>``;
export const SafeAreaView = styled.SafeAreaView<ViewProps>``;

interface RowProps {
  backgroundColor?: string;
  width?: numberStringNull;
  alignItems?: string;
  justifyContent?: string;
  paddingHorizontal?: number;
  paddingVertical?: number;
  marginTop?: number;
  marginBottom?: number;
  marginRight?: number;
  flex?: number | string;
}

export const Row = styled.View<RowProps>`
  background-color: ${({ backgroundColor = 'transparent' }) => backgroundColor};
  width: ${({ width = 'auto' }) => width};
  flex-direction: row;
  align-items: ${({ alignItems = 'flex-start' }) => alignItems};
  justify-content: ${({ justifyContent = 'flex-start' }) => justifyContent};
  padding-horizontal: ${({ paddingHorizontal = 0 }) => paddingHorizontal}px;
  padding-vertical: ${({ paddingVertical = 0 }) => paddingVertical}px;
  margin-top: ${({ marginTop = 0 }) => marginTop}px;
  margin-bottom: ${({ marginBottom = 0 }) => marginBottom}px;
  margin-right: ${({ marginRight = 0 }) => marginRight}px;
  ${({ flex }) =>
    flex &&
    css`
      flex: ${flex};
    `}
`;

export const FlatList = styled.FlatList.attrs({
  showsVerticalScrollIndicator: false,
})``;

export const Flex = styled.View<{ justifyContent?: string }>`
  flex: 1;
  width: 100%;
`;

export const SafeBackground = styled.SafeAreaView`
  background-color: #ffffff;
  flex: 1;
`;

export const Background = styled.View<{ paddingHorizontal?: number }>`
  background-color: #ffffff;
  padding-horizontal: ${(props) => props?.paddingHorizontal}px;
`;

export const Scroll = styled.ScrollView.attrs({
  showsVerticalScrollIndicator: false,
})<{ backgroundColor?: string }>`
  flex: 1;
  background-color: ${({ backgroundColor = colors.white }) => backgroundColor};
`;

export const EndCenter = styled.View`
  align-items: center;
  justify-content: flex-end;
`;

export interface IconBaseProps extends ImageProps {
  height?: number;
  width?: number;
  size?: number;
}

export const IconBase = styled(FastImage).attrs(({ resizeMode }: { resizeMode?: string }) => ({
  resizeMode,
}))<IconBaseProps>`
  height: ${({ size = 18, height }: IconBaseProps) => scaleFontSize(height ?? size)}px;
  width: ${({ size = 18, width }: IconBaseProps) => scaleFontSize(width ?? size)}px;
`;

export interface IconProps {
  resizeMode?: string;
  containerStyle?: StyleProp<ViewStyle>;
  style?: StyleProp<ViewStyle>;
  onPress?: any;
  testID?: string;
  source?: ImageSourcePropType;
  size?: number;
  imageProps?: IconBaseProps;
  containerProps?: TouchableOpacityProps;
}

export const Icon = ({
  resizeMode = 'contain',
  containerStyle = {},
  style = {},
  onPress = null,
  testID = '',
  source,
  size = 18,
  imageProps = {},
  containerProps = {},
}: IconProps) => (
  <TouchableOpacity
    disabled={!onPress}
    onPress={onPress}
    style={containerStyle}
    testID={testID}
    {...containerProps}
  >
    <IconBase resizeMode={resizeMode} source={source} style={style} size={size} {...imageProps} />
  </TouchableOpacity>
);

interface DefaultBackgroundProps {
  safestyle?: StyleProp<ViewStyle>;
  style?: StyleProp<ViewStyle>;
  children: React.ReactNode;
  isLoading?: boolean;
  paddingHorizontal?: number;
  testID?: string;
}

export const DefaultBackground = ({
  safestyle = {},
  style = {},
  children,
  isLoading = false,
  paddingHorizontal = 16,
  testID = '',
}: DefaultBackgroundProps) => (
  <>
    <SafeBackground testID={testID} style={safestyle}>
      <Background style={style} paddingHorizontal={paddingHorizontal}>
        {children}
      </Background>
    </SafeBackground>
    <LoadingOverlay visible={isLoading} />
  </>
);

interface IDefaultScrollBackgroundProps
  extends Pick<
    DefaultBackgroundProps,
    'safestyle' | 'style' | 'children' | 'isLoading' | 'testID'
  > {
  contentContainerStyle?: StyleProp<ViewStyle>;
  scrollComponent?: React.ComponentType<any> | null;
}

export const DefaultScrollBackground = ({
  safestyle = {},
  style = {},
  children,
  isLoading = false,
  contentContainerStyle = {},
  testID = '',
  scrollComponent = null,
}: IDefaultScrollBackgroundProps) => {
  const ScrollContainer = scrollComponent ?? Scroll;
  return (
    <>
      <SafeBackground testID={testID} style={safestyle}>
        <ScrollContainer
          testID="scrollView"
          style={style}
          contentContainerStyle={contentContainerStyle}
        >
          {children}
        </ScrollContainer>
      </SafeBackground>
      <LoadingOverlay visible={isLoading} />
    </>
  );
};

export const LoadingOverlay = ({ visible }: { visible: boolean }) => (
  <Spinner
    visible={visible}
    overlayColor="rgba(255, 255, 255, 0.70)"
    testID="loadingOverlayScreen"
    customIndicator={
      <LottieView
        style={globalStyles.lottieView}
        source={require('../../assets/loading-animation.json')}
        autoPlay
        loop
      />
    }
  />
);

export const LoadingIndicator = () => (
  <Flex justifyContent="center">
    <ActivityIndicator />
  </Flex>
);

interface ISeparatorProps {
  width?: number | string;
  backgroundColor?: string;
  height?: number;
  marginLeft?: number;
  marginTop?: number;
  marginHorizontal?: number;
  marginBottom?: number;
}

export const Separator = styled.View<ISeparatorProps>`
  width: ${({ width = '100%' }) => width};
  background-color: ${({ backgroundColor = colors.text5 }) => backgroundColor};
  height: ${({ height = 1 }) => height}px;
  margin-left: ${({ marginLeft = 0 }) => marginLeft}px;
  margin-top: ${({ marginTop = 0 }) => marginTop}px;
  margin-horizontal: ${({ marginHorizontal = 0 }) => marginHorizontal}px;
  margin-bottom: ${({ marginBottom = 0 }) => marginBottom}px;
`;

interface ISelectionProps {
  isSelected?: boolean;
  marginTop?: number;
  marginHorizontal?: number;
  paddingHorizontal?: number;
}
export const Selection = styled.TouchableOpacity<ISelectionProps>`
  flex-direction: row;
  background-color: ${({ isSelected = false }) => (isSelected ? colors.white : '#f9f9f9')};
  margin-top: ${({ marginTop = 8 }) => marginTop}px;
  margin-horizontal: ${({ marginHorizontal = 16 }) => marginHorizontal}px;
  padding-horizontal: ${({ paddingHorizontal = 16 }) => paddingHorizontal}px;
  border-radius: 6px;
  border: ${({ isSelected = false }) => (isSelected ? '1px solid #16C793' : '#f9f9f9')}
  justify-content: space-between;
  align-items: center;
  min-height: ${scaleFontSize(48)}px;
`;

interface IColumnProps {
  width?: numberStringNull;
  backgroundColor?: string;
  alignItems?: string;
  justifyContent?: string;
  padding?: number;
  paddingHorizontal?: number;
  paddingVertical?: number;
  marginTop?: number;
  marginBottom?: number;
  flex?: number | string;
}

export const Column = styled.View<IColumnProps>`
  width: ${({ width = 'auto' }) => width};
  background-color: ${({ backgroundColor = 'transparent' }) => backgroundColor};
  align-items: ${({ alignItems = 'flex-start' }) => alignItems};
  justify-content: ${({ justifyContent = 'flex-start' }) => justifyContent};
  padding-horizontal: ${({ paddingHorizontal = 0, padding = 0 }) => padding ?? paddingHorizontal}px;
  padding-vertical: ${({ paddingVertical = 0, padding = 0 }) => padding ?? paddingVertical}px;
  margin-top: ${({ marginTop = 0 }) => marginTop}px;
  margin-bottom: ${({ marginBottom = 0 }) => marginBottom}px;
  ${({ flex }) =>
    flex &&
    css`
      flex: ${flex};
    `}
`;

export const globalStyles = StyleSheet.create({
  flex: {
    flex: 1,
  },
  marginTop0: {
    marginTop: 0,
  },
  marginTop8: {
    marginTop: 8,
  },
  marginTop16: {
    marginTop: 16,
  },
  marginBottom8: {
    marginBottom: 8,
  },
  marginBottom16: {
    marginBottom: 16,
  },
  marginLeft8: {
    marginLeft: 8,
  },
  marginLeft12: {
    marginLeft: 12,
  },
  marginLeft16: {
    marginLeft: 16,
  },
  marginRight8: {
    marginRight: 8,
  },
  marginRight16: {
    marginRight: 16,
  },
  marginRight32: {
    marginRight: 32,
  },
  paddingHorizontal16: {
    paddingHorizontal: 16,
  },
  paddingRight16: {
    paddingRight: 16,
  },
  paddingTop16: {
    paddingTop: 16,
  },
  withNotch: {
    flex: 1,
    backgroundColor: colors.white,
    paddingTop: hasNotch() ? 16 : 24,
  },
  width100: {
    width: '100%',
  },
  bgColorWhite: {
    backgroundColor: colors.white,
  },
  bgColorBackground: {
    backgroundColor: colors.background,
  },
  flexbgColorBackground: {
    flex: 1,
    backgroundColor: colors.background,
  },
  alignCenter: {
    alignItems: 'center',
  },
  lottieView: {
    height: 100,
    width: 100,
  },
  headerScrollTitle: {
    marginLeft: 0,
    paddingLeft: 16,
    paddingVertical: 4,
    fontFamily: fontFamilies.demiBold,
    fontSize: scaleFontSize(24),
    fontWeight: '600',
  },
});
