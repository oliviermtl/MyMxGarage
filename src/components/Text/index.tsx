import { DefaultView, TouchableOpacity } from '../global';
import { stringToRegex } from '../../utils/helper';
import { StyleProp, TextStyle, ViewStyle } from 'react-native';
import Hyperlink from 'react-native-hyperlink';
import {
  H1,
  H2,
  H3,
  H4,
  P1,
  P2,
  P3,
  C1,
  C2,
  Form,
  Button1,
  Button2,
  Button3,
  ParsedText,
} from './components';
import { colors } from '../../constants';

export type TextTypes =
  | 'H1'
  | 'H2'
  | 'H3'
  | 'H4'
  | 'P1'
  | 'P2'
  | 'P3'
  | 'C1'
  | 'C2'
  | 'Button1'
  | 'Button2'
  | 'Button3';

export type AlignTypes = 'left' | 'center' | 'right';

export interface TextProps {
  type?: TextTypes;
  style?: StyleProp<TextStyle>;
  children?: React.ReactNode;
  width?: null;
  align?: AlignTypes;
  color?: string;
  marginTop?: number;
  paddingLeft?: number;
  paddingHorizontal?: number;
  numberOfLines?: number | null;
  fontFamily?: string | null;
  testID?: string | null;
}

export const Text = ({
  type = 'P1',
  style = {},
  children = null,
  width,
  align = 'left',
  color = colors.offBlack,
  marginTop = 0,
  paddingLeft = 0,
  paddingHorizontal = 0,
  numberOfLines = null,
  fontFamily = null,
  testID = 'text',
}: TextProps) => {
  const TextType = { H1, H2, H3, H4, P1, P2, P3, C1, C2, Form, Button1, Button2, Button3 }[type];
  return (
    <TextType
      align={align}
      style={style}
      width={width}
      color={color}
      marginTop={marginTop}
      paddingLeft={paddingLeft}
      paddingHorizontal={paddingHorizontal}
      numberOfLines={numberOfLines}
      fontFamily={fontFamily}
      testID={testID}
    >
      {children}
    </TextType>
  );
};

export interface PatternStyledTextProps {
  text: string;
  pattern?: any;
  onPress?: any;
  containerStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  textColor?: string;
  patternStyle?: StyleProp<TextStyle>;
  testID?: string | null;
}
export const PatternStyledText = ({
  text,
  pattern = null,
  onPress = null,
  containerStyle = {},
  textStyle = {},
  textColor = '#000',
  patternStyle = {},
  testID = 'pattern',
}: PatternStyledTextProps) => {
  const Container = onPress ? TouchableOpacity : DefaultView;
  return (
    <Container onPress={onPress} style={containerStyle} testID={testID}>
      <ParsedText
        style={textStyle}
        color={textColor}
        parse={[
          {
            pattern: pattern || stringToRegex(text),
            style: patternStyle,
          },
        ]}
      >
        {text}
      </ParsedText>
    </Container>
  );
};

interface HyperlinkTextProps {
  text: React.ReactNode;
  linkStyle?: StyleProp<TextStyle>;
  textStyle?: StyleProp<TextStyle>;
  type?: TextTypes;
  width?: null;
  align?: AlignTypes;
  color?: string;
  marginTop?: number;
  paddingLeft?: number;
  paddingHorizontal?: number;
  onPress?: any;
  onLongPress?: any;
  linkText?: ((url: string) => string) | string;
}
export const HyperlinkText = ({
  text,
  linkStyle = {},
  textStyle = {},
  type = 'P2',
  width,
  align = 'left',
  color = colors.text2,
  marginTop = 0,
  paddingLeft = 0,
  paddingHorizontal = 0,
  onPress = null,
  onLongPress = null,
  linkText = '',
}: HyperlinkTextProps) => (
  <Hyperlink
    linkText={linkText}
    onPress={onPress}
    onLongPress={onLongPress}
    linkStyle={[{ color: colors.wildTide, textDecorationLine: 'underline' }, linkStyle]}
  >
    <Text
      width={width}
      type={type}
      align={align}
      color={color}
      marginTop={marginTop}
      paddingHorizontal={paddingHorizontal}
      paddingLeft={paddingLeft}
      style={textStyle}
    >
      {text}
    </Text>
  </Hyperlink>
);
