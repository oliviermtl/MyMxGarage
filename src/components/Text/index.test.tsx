import { fireEvent, render } from '@testing-library/react-native/pure';
import { Text, PatternStyledText, HyperlinkText } from 'components/Text';

describe('Text', () => {
  it('renders Text Snapshot', () => {
    const tree = render(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text C1', () => {
    const tree = render(<Text type="C1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with color black', () => {
    const tree = render(<Text color="black" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with margin Top 32', () => {
    const tree = render(<Text marginTop={32} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with width null', () => {
    const tree = render(<Text width={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with align right', () => {
    const tree = render(<Text align="right" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with align default', () => {
    const tree = render(<Text align={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with color default', () => {
    const tree = render(<Text color={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with paddingHorizontal 16', () => {
    const tree = render(<Text paddingHorizontal={16} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Text with paddingLeft 16', () => {
    const tree = render(<Text paddingLeft={16} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders PatternStyledText Snapshot', () => {
    const tree = render(<PatternStyledText />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders PatternStyledText onPress', () => {
    const mockFn = jest.fn();
    const { getByText } = render(<PatternStyledText text="test" onPress={mockFn} />);
    fireEvent.press(getByText('test'));
    expect(mockFn).toHaveBeenCalled();
  });

  it('renders HyperlinkText Snapshot', () => {
    const tree = render(<HyperlinkText />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders HyperlinkText onPress', () => {
    const mockFn = jest.fn();
    const { getByText } = render(<HyperlinkText text="test" onPress={mockFn} />);
    fireEvent.press(getByText('test'));
    expect(mockFn).toHaveBeenCalled();
  });
});
