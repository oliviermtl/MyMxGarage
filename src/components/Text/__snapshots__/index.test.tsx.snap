// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders HyperlinkText Snapshot 1`] = `
<View
  injectViewProps={[Function]}
>
  <Text
    align="left"
    color="#222732"
    fontFamily={null}
    marginTop={0}
    numberOfLines={null}
    paddingHorizontal={0}
    paddingLeft={0}
    style={
      Array [
        Object {
          "color": "#222732",
          "fontFamily": "Mulish-Regular",
          "fontSize": 28,
          "lineHeight": 40,
          "marginTop": 0,
          "textAlign": "left",
          "width": "100%",
        },
        Object {},
      ]
    }
    testID="text"
  />
</View>
`;

exports[`Text renders PatternStyledText Snapshot 1`] = `
<View
  onPress={null}
  style={
    Array [
      Object {},
      Object {},
    ]
  }
  testID="pattern"
>
  <Text
    color="#000"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Mulish-Regular",
          "fontSize": 28,
          "letterSpacing": 0.2,
          "textAlign": "center",
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Text renders Text C1 1`] = `
<Text
  align="left"
  color="#0D1119"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 26,
        "lineHeight": 36,
        "marginTop": 0,
        "textAlign": "left",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text Snapshot 1`] = `
<Text
  align="left"
  color="#0D1119"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "textAlign": "left",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with align default 1`] = `
<Text
  align={null}
  color="#0D1119"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "textAlign": "",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with align right 1`] = `
<Text
  align="right"
  color="#0D1119"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "textAlign": "right",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with color black 1`] = `
<Text
  align="left"
  color="black"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "textAlign": "left",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with color default 1`] = `
<Text
  align="left"
  color={null}
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "textAlign": "left",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with margin Top 32 1`] = `
<Text
  align="left"
  color="#0D1119"
  fontFamily={null}
  marginTop={32}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 32,
        "textAlign": "left",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with paddingHorizontal 16 1`] = `
<Text
  align="left"
  color="#0D1119"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={16}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "textAlign": "left",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with paddingLeft 16 1`] = `
<Text
  align="left"
  color="#0D1119"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={16}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "paddingLeft": 16,
        "textAlign": "left",
        "width": "100%",
      },
      Object {},
    ]
  }
  testID="text"
/>
`;

exports[`Text renders Text with width null 1`] = `
<Text
  align="left"
  color="#0D1119"
  fontFamily={null}
  marginTop={0}
  numberOfLines={null}
  paddingHorizontal={0}
  paddingLeft={0}
  style={
    Array [
      Object {
        "color": "#0D1119",
        "fontFamily": "Mulish-Regular",
        "fontSize": 32,
        "lineHeight": 44,
        "marginTop": 0,
        "textAlign": "left",
        "width": "auto",
      },
      Object {},
    ]
  }
  testID="text"
  width={null}
/>
`;
